terraform/environments/data-platform/iam.tf

```
@@ -314,11 +507,13 @@ data "aws_iam_policy_document" "iam_policy_document_for_create_metadata_lambda"
 }
 
 data "aws_iam_policy_document" "iam_policy_document_for_reload_data_product_lambda" {
+  source_policy_documents = [data.aws_iam_policy_document.log_to_bucket.json, data.aws_iam_policy_document.read_metadata.json]
+
   statement {
     sid       = "ListBucket"
     effect    = "Allow"
     actions   = ["s3:ListBucket"]
-    resources = [module.s3-bucket.bucket.arn, "${module.s3-bucket.bucket.arn}/*"]
+    resources = [module.data_s3_bucket.bucket.arn, "${module.data_s3_bucket.bucket.arn}/*"]
```

```suggestion
    resources = [module.data_s3_bucket.bucket.arn]
```
This is a ListBucket operation, so only functions on the bucket object, not key paths